Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/clip.h

```diff
@@ -633,119 +635,195 @@ clip(TriangleMesh& tm,
 /**
   * \ingroup PMP_corefinement_grp
   *
-  * \brief clips `tm` by keeping the part that is on the negative side of `plane` (side opposite to its normal vector).
+  * \brief clips `pm` by keeping the part that is on the negative side of `plane` (side opposite to its normal vector).
   *
-  * If `tm` is closed, the clipped part can be closed too if the named parameter `clip_volume` is set to `true`.
+  * If `pm` is closed, the clipped part can be closed too if the named parameter `clip_volume` is set to `true`.
   * See Subsection \ref coref_clip for more details.
   *
-  * \note `Plane_3` must be from the same %Kernel as the point of the internal vertex point map of `TriangleMesh`.
-  * \note `Plane_3` must be from the same %Kernel as the point of the vertex point map of `tm`.
+  * \note `Plane_3` must be from the same kernel as the point of the internal vertex point map of `PolygonMesh`.
+  * \note `Plane_3` must be from the same kernel as the point of the vertex point map of `pm`.
   *
-  * \pre \link CGAL::Polygon_mesh_processing::does_self_intersect() `!CGAL::Polygon_mesh_processing::does_self_intersect(tm)` \endlink
+  * \pre \link CGAL::Polygon_mesh_processing::does_self_intersect() `!CGAL::Polygon_mesh_processing::does_self_intersect(pm)` \endlink
```

Comment:
not defined if pm is not triangulated. Just remove anyway since we now have `NP::allow_self_intersections` ?